app/controllers/songs_controller.rb

```diff
@@ -0,0 +1,46 @@
+class SongsController < ApplicationController
+
+  def index
+
+  end
+
+  def new
+
+  end
+
+  def show
+    client = Grooveshark::Client.new({session: session[:groove_session]})
+    @groove_song = Song.find(params[:id])
+    @groove_url = client.get_song_url_by_id(@groove_song.link)
```

Comment:
I think you should have a model or service object that handles the mapping of Song to groove shark link.  We can discuss this more tomorrow.